src/network/connection_dispatcher_task.cpp

```diff
@@ -10,19 +10,25 @@
 namespace terrier::network {
 
 ConnectionDispatcherTask::ConnectionDispatcherTask(
-    uint32_t num_handlers, int listen_fd, common::DedicatedThreadOwner *dedicated_thread_owner,
+    uint32_t num_handlers, common::DedicatedThreadOwner *dedicated_thread_owner,
     common::ManagedPointer<ProtocolInterpreter::Provider> interpreter_provider,
     common::ManagedPointer<ConnectionHandleFactory> connection_handle_factory,
-    common::ManagedPointer<common::DedicatedThreadRegistry> thread_registry)
+    common::ManagedPointer<common::DedicatedThreadRegistry> thread_registry,
+    std::initializer_list<int> file_descriptors)
     : NotifiableTask(MASTER_THREAD_ID),
       num_handlers_(num_handlers),
       dedicated_thread_owner_(dedicated_thread_owner),
       connection_handle_factory_(connection_handle_factory),
       thread_registry_(thread_registry),
       interpreter_provider_(interpreter_provider),
       next_handler_(0) {
-  RegisterEvent(listen_fd, EV_READ | EV_PERSIST, METHOD_AS_CALLBACK(ConnectionDispatcherTask, DispatchConnection),
-                this);
+  for (int listen_fd : file_descriptors) {
+    if (listen_fd >= 0) {
```

Comment:
That's intentional. Do you want me to change this? If I do, then it'd make sense to change how we handle invalid file descriptors elsewhere. 

We already warn (and provide some nice information) whenever we fail to get a file descriptor. Whenever that happens, though, we handle it gracefully: it's possible that we failed to get a Unix domain socket but still get a network socket, or vice versa. In those cases, we keep the server up. This is vaguely similar to what Postgres does, where it may keep listening to a network socket even if it fails to get a Unix domain socket.

Reply:
Hm. I guess I am OK with this for now (to limit the PR's scope, and because errors are unlikely to happen in practice). I am in favor of the caller swallowing the exception and doing nothing, since there are many fd's getting passed in and I am uneasy about the semantics of "some items in the list will work and some won't".